### ISSUE TITLE
Reverse Youtube Playlist

### ISSUE THREAD

**Issue by CyberMew** [YePpHa/YouTubeCenter#396]
A lot of the playlist go in descending order than ascending order, and this makes it hard to follow a video series in chronological order.

Something like this would be excellent: https://chrome.google.com/webstore/detail/reverse-youtube-playlist/ajhonbaagcobjdmbocblbebcmbmmbfmi?hl=en

Also, +1 to expand volume bar by default!

**Comment by Eisys** [comment]
Hey @Yonezpt Could you also try to script this request, a simple reverse playlist feature. It has been requested alot and I myself use the addon mentioned and it would be one less addon needed besides YTC. 
Maybe this is difficult, considering @YePpHa  hasn't implemented it yet but if you'd be up for it, I and probably few others would greatly appreciate it!

**Comment by Yonezpt** [comment]
I've played with it a while back, didn't find it difficult, will try to give it a go once I am finished with the other one.

**Comment by Eisys** [comment]
Ok sweet, thanks! Hope you succeed!

**Comment by Yonezpt** [comment]
Alright, sorry for the delay, but time flew by and I had to call a timeout on this script yesterday.

I have successfully created a fully functional reverse playlist script which, as far as I could see, has no rival. All the other extensions that exist for the same purpose are either broken and/or only work partially. This one remembers the last playlist you reversed and works in fullscreen as well (at least it has been working perfectly for me).

![ss](https://cloud.githubusercontent.com/assets/8498057/5817377/cb35cfea-a09a-11e4-87b6-73026f7367ac.png)

How it works is pretty simple; A new button will show in any playlist you open (including the youtube generated playlists, aka "radio playlists"), pressing it will reverse the playlist order and remember that the current list is to stay reversed until the user clicks on the button again.

Here it is, give it a try and let me know if you found any issues.

https://openuserjs.org/scripts/ParticleCore/Reverse_Playlist_for_YouTube

**Comment by Eisys** [comment]
Yup, it's perfect and it works perfectly for me so far, in fullscreen too! And I really like the icon for it.

Thank you so much @Yonezpt !!

Let's hope @YePpHa adds this as well!

**Comment by Yonezpt** [comment]
@Eisys While trying to find when I played with the reverse playlist I noticed that it was because of you in this comment: https://github.com/YePpHa/YouTubeCenter/issues/1013#issuecomment-59388704
I started digging around the code after that, but I never finished it by then because I assumed YePpHa would code it in no time and I also didn't have the knowledge I have today.

**Comment by Yonezpt** [comment]
@Eisys I have changed the name a bit because it was causing problems with another addon that had the exact name, I have already updated the link in my previous reply for you to use it. I doubt there is anything I might change in the near future, but it is best to use the one that can be updated whenever it needs to.

**Comment by Eisys** [comment]
Ok I updated it!
Are you expecting @YePpHa to add it to YTC or do you think it'll have to be a seperate script?

**Comment by Yonezpt** [comment]
I have no idea, it is up to him, but since he hasn't refused it when people requested the feature I assume that he will implement it in YTC as soon as he is able to. I just code these things independently because people, like you, asked or because someone just wants that single feature, but these scripts are also for YePpHa if he finds them helpful in any way.

**Comment by Eisys** [comment]
Ah ok, that's cool!
Have you ever thought about working with YePpHa on YTC? It seems that you have alot more time than he does these days, and you seem equally as talented!

**Comment by Yonezpt** [comment]
I did, I already ventured once before, but his code is confusing for me and I'd cause more problems than not, so I prefer to just help the way I can and he adapts the code to his own which he knows well. There's also certain details that shouldn't be changed otherwise it might mess the compiler or the ANT system, or something else.

**Comment by ronCYA** [comment]
What is the progress/prospect of this happening at the moment? And which of these is the 'official' request?:

#104
#396
#955

**Comment by FichteFoll** [comment]
Awesome, this works likea charm.

Usually the 'official' request is the oldest, unless a newer one has more information in it (which is the case here, since we actually have a solution).

**Comment by ireun** [comment]
Yup, and since this one has the most infos in it, I've closed #104.

**Comment by ronCYA** [comment]
Cukie smookie! Too easy.

Now just fingers crossed this'll be implemented one day.

**Comment by TalonDarkclaw** [comment]
Thank you, Thank you, Thank you for this script.
I can't tell you have unbelieveabley helpful and awesome it is.
Thank you.
Thank you.
Thank you.

**Comment by bigworld12** [comment]
guys if you can combine these two extensions 
https://chrome.google.com/webstore/detail/youtube-playlist-load-all/pbclhlmcclobinpephoflilgclodhnmf?hl=en

and

https://chrome.google.com/webstore/detail/reverse-watch-later-list/fghdmhmlgblfeojfcejdoijadldnhend?hl=en

would be great ^^

**Comment by Sja91** [comment]
So this has been requested for this long and nothing new about it?
The issue in the first post is really the biggest reason we need this feature, It is a very common issue, and shouldn't even be that hard to implement.
It can easily be done all on the client side, throw a flag in the URL that shows the playlist needs to play the last video instead of the next video, then the client just has to load the last video instead of the next on the list.
This would take all of a few lines of code.

**Comment by ErdeFB** [comment]
Active development of YTC has ceased.
